jobs/statistics/ProcessUsageStatsLogFile.php

```diff
@@ -0,0 +1,246 @@
+        $this->loadId = $loadId;
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle(): void
+    {
+        $filename = $this->loadId;
+        $dispatchFilePath = StatisticsHelper::getUsageStatsDirPath() . '/' . FileLoader::FILE_LOADER_PATH_DISPATCH . '/' . $filename;
+        if (!file_exists($dispatchFilePath)) {
+            throw new JobException(__(
+                'admin.job.processLogFile.fileNotFound',
+                ['file' => $dispatchFilePath]
+            ));
+        }
+        $this->process($dispatchFilePath);
+    }
+
+
+    protected function process(string $dispatchFilePath): void
+    {
+        $fhandle = fopen($dispatchFilePath, 'r');
+        if (!$fhandle) {
+            // reject file -- move the file from dispatch to reject folder
+            $filename = $this->loadId;
+            $rejectFilePath = StatisticsHelper::getUsageStatsDirPath() . '/' . FileLoader::FILE_LOADER_PATH_REJECT . '/' . $filename;
+            if (!rename($dispatchFilePath, $rejectFilePath)) {
+                $message = __('admin.job.compileMetrics.returnToStaging.error', ['file' => $filename,
+                    'dispatchFilePath' => $dispatchFilePath, 'rejectFilePath' => $rejectFilePath]);
+                error_log($message);
+            }
+            throw new JobException(__('admin.job.processLogFile.openFileFailed', ['file' => $dispatchFilePath]));
+        }
+
+        // Make sure we don't have any temporary records associated
+        // with the current load ID in database.
+        $this->deleteByLoadId();
+
+        $lineNumber = 0;
+        while (!feof($fhandle)) {
+            $lineNumber++;
+            $line = trim(fgets($fhandle));
+            if (empty($line) || substr($line, 0, 1) === '#') {
+                continue;
+            } // Spacing or comment lines. This actually should not occur in the new format.
+
+            $entryData = json_decode($line);
+            if ($entryData === null) {
+                // This line is not in the right format.
+                $message = __(
+                    'admin.job.processLogFile.wrongLoglineFormat',
+                    ['file' => $this->loadId, 'lineNumber' => $lineNumber]
+                );
+                error_log($message);
+                continue;
+            }
+
+            try {
+                $this->isLogEntryValid($entryData);
+            } catch (Exception $e) {
+                $message = __(
+                    'admin.job.processLogFile.invalidLogEntry',
+                    ['file' => $this->loadId, 'lineNumber' => $lineNumber, 'error' => $e->getMessage()]
+                );
+                error_log($message);
+                continue;
+            }
+
+            // Avoid bots.
+            if (Core::isUserAgentBot($entryData->userAgent)) {
+                continue;
+            }
+
+            $this->insertTemporaryUsageStatsData($entryData, $lineNumber);
+        }
+        fclose($fhandle);
```

Comment:
Not a big deal, but it's safer to use:
```php
fopen();
try{
  ...
} finally {
  fclose();
}
```

You might also use the [SplFileObject](https://www.php.net/splfileobject).

Reply:
I will then use SplFileObject...